src/systems/systems/src/pressurization/mod.rs

```diff
@@ -0,0 +1,952 @@
+    fn should_close_pressure_valve(&self) -> bool;
+    fn target_valve_position(&self) -> Ratio;
+}
+
+pub struct AircraftInputsPressurization {
+    eng_1_n1: Ratio,
+    eng_2_n1: Ratio,
+    landing_elevation: Length,
+    sea_level_pressure: Pressure,
+    destination_qnh: Pressure,
+}
+
+impl AircraftInputsPressurization {
+    fn new() -> Self {
+        Self {
+            eng_1_n1: Ratio::new::<percent>(0.),
+            eng_2_n1: Ratio::new::<percent>(0.),
+            landing_elevation: Length::new::<foot>(0.),
+            sea_level_pressure: Pressure::new::<hectopascal>(1013.),
+            destination_qnh: Pressure::new::<hectopascal>(1013.),
+        }
+    }
+
+    pub fn set_eng_n1(&mut self, eng_1_n1: Ratio, eng_2_n1: Ratio) {
+        self.eng_1_n1 = eng_1_n1;
+        self.eng_2_n1 = eng_2_n1;
+    }
+
+    pub fn set_landing_elev(&mut self, landing_elev: Length) {
+        self.landing_elevation = landing_elev;
+    }
+
+    pub fn set_sea_level_pressure(&mut self, sea_level_pressure: Pressure) {
+        self.sea_level_pressure = sea_level_pressure;
+    }
+
+    pub fn set_destination_qnh(&mut self, dest_qnh: Pressure) {
+        self.destination_qnh = dest_qnh;
+    }
+
+    pub fn eng_1_n1(&self) -> Ratio {
+        self.eng_1_n1
+    }
+
+    pub fn eng_2_n1(&self) -> Ratio {
+        self.eng_2_n1
+    }
+
+    pub fn landing_elev(&self) -> Length {
+        self.landing_elevation
+    }
+
+    pub fn sea_level_pressure(&self) -> Pressure {
+        self.sea_level_pressure
+    }
+
+    pub fn destination_qnh(&self) -> Pressure {
+        self.destination_qnh
+    }
+}
+
+pub struct Pressurization {
+    cpc_1: CabinPressureController,
+    cpc_2: CabinPressureController,
+    outflow_valve: PressureValve,
+    // safety_valve: PressureValve,
+    // rcpu: bool,
+    // is_in_man_mode: bool,
+    // man_mode_duration: Duration,
+    active_system: usize,
+    aircraft_inputs: AircraftInputsPressurization,
+}
+
+impl Pressurization {
+    pub fn new() -> Self {
+        let random = random_number();
+        let mut active: usize = 1;
+        if random % 2 == 0 {
+            active = 2
+        }
+
+        Self {
+            cpc_1: CabinPressureController::new(),
+            cpc_2: CabinPressureController::new(),
+            outflow_valve: PressureValve::new(),
+            // safety_valve: PressureValve::new(),
+            // rcpu:
+            // is_in_man_mode: false,
+            // man_mode_duration: Duration::from_millis(0),
+            active_system: active,
+            aircraft_inputs: AircraftInputsPressurization::new(),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        press_overhead: &PressurizationOverheadPanel,
+        eng_1_n1: Ratio,
+        eng_2_n1: Ratio,
+    ) {
+        self.aircraft_inputs.set_eng_n1(eng_1_n1, eng_2_n1);
+
+        if !self.is_sys1_active() && !self.is_sys2_active() {
+            self.set_active_system();
+        };
```

Comment:
**question**: In what situation might both become inactive?

Reply:
I've removed this now, however both systems _could_ be inactive if there was a fault in both at the same time, or if full manual mode was engaged. In any case this will be the scope for a future PR! 😄